utils/message-generator/src/executor.rs

```diff
@@ -339,50 +331,51 @@ impl Executor {
                                 },
                                 Ok(roles_logic_sv2::parsers::TemplateDistribution::SetNewPrevHash(m)) => {
                                     if message_type.as_str() == "SetNewPrevHash" {
                                         let msg = serde_json::to_value(&m).unwrap();
-                                        check_msg_field(msg,&field_name,&value_type,field);
+                                        check_each_field(msg,field_data);
                                     }
                                 },
                                 Ok(roles_logic_sv2::parsers::TemplateDistribution::CoinbaseOutputDataSize(m)) => {
                                     if message_type.as_str() == "CoinbaseOutputDataSize" {
                                         let msg = serde_json::to_value(&m).unwrap();
-                                        check_msg_field(msg,&field_name,&value_type,field);
+                                        check_each_field(msg,field_data);
                                     }
                                 },
                                 Ok(roles_logic_sv2::parsers::TemplateDistribution::RequestTransactionData(m)) => {
                                     if message_type.as_str() == "RequestTransactionData" {
                                         let msg = serde_json::to_value(&m).unwrap();
-                                        check_msg_field(msg,&field_name,&value_type,field);
+                                        check_each_field(msg,field_data);
                                     }
                                 },
                                 Ok(roles_logic_sv2::parsers::TemplateDistribution::RequestTransactionDataError(m)) => {
                                     if message_type.as_str() == "RequestTransactionDataError" {
                                         let msg = serde_json::to_value(&m).unwrap();
-                                        check_msg_field(msg,&field_name,&value_type,field);
+                                        check_each_field(msg,field_data);
                                     }
                                 },
                                 Ok(roles_logic_sv2::parsers::TemplateDistribution::RequestTransactionDataSuccess(m)) => {
                                     if message_type.as_str() == "RequestTransactionDataSuccess" {
                                         let msg = serde_json::to_value(&m).unwrap();
-                                        check_msg_field(msg,&field_name,&value_type,field);
+                                        check_each_field(msg,field_data);
                                     }
                                 },
                                 Err(e) => panic!("err {:?}",e),
                             }
                         } else {
-                            panic!()
+                            panic!("match_message_field subprotocol not valid")
```

Comment:
Would be useful to spit out the subprotocol value in the panic to make debugging easier.